contract/bridge/contracts/ExchangeRateOracle.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.12;
+
+import {ICollateral} from "./Collateral.sol";
+
+abstract contract ExchangeRateOracle is ICollateral {
+    uint128 constant MAX_DELAY = 1000;
+    uint128 public last_exchange_rate_time;
```

Comment:
looks good. just minor coding styles like, instead of using underscores, use mixedCase: https://docs.soliditylang.org/en/v0.5.3/style-guide.html

Reply:
I also use some underscores, because the [btc-parachain](https://github.com/interlay/btc-parachain.git) use underscores, it can be easy to find the corresponding code.

Reply:
I actually decided to use underscores to be consistent with @peekpi 's existing contracts. Would be good to have consensus here. We should stick to the official style guide.

Reply:
@gupadhyaya ready for a final review before merge. Cleaned up syntax to be in line with Solidity best practices.